icechunk/src/change_set.rs

```diff
@@ -151,14 +141,24 @@ impl ChangeSet {
         self.updated_attributes.remove(node_id);
         self.set_chunks.remove(node_id);
         if !is_new_array {
-            self.deleted_arrays.insert(path);
+            self.deleted_arrays.insert((path, node_id.clone()));
         }
     }
 
-    pub fn is_deleted(&self, path: &Path) -> bool {
-        self.deleted_groups.contains(path)
-            || self.deleted_arrays.contains(path)
-            || path.ancestors().skip(1).any(|parent| self.is_deleted(&parent))
+    pub fn is_deleted(&self, path: &Path, node_id: &NodeId) -> bool {
+        dbg!(format!("checking is_deleted for {0} & {1}", path, node_id));
+        let key = (path.clone(), node_id.clone());
+        self.deleted_groups.contains(&key)
+            || self.deleted_arrays.contains(&key)
+            || path.ancestors().skip(1).any(|parent| {
+                self.get_group(&parent).is_some_and(|parent_node_id| {
```

Comment:
This is wrong, and an existing test fails. 

We are in `ChangeSet` and somehow need to know the `NodeId` for a node that is in the snapshot. How do I get that?

Reply:
:thinking:  ... I think the problem is  with the `get_group` call right? That's what doesn't work... but it feels this is fixable? We need to know if a given path was deleted _in this changeset_, we shouldn't need the repo for that.

Reply:
We need to know if a *node* was deleted, not just the path (that's the issue :) )